### PR TITLE
Removed MXNET_USE_MKLDNN definition when option USE_MKLDNN is disabled

### PR DESCRIPTION
**Pull request by lebeg**
## Description ##

Builds were failing to wrong include.

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)

### Changes ###
- [x] Added condition for MXNET_USE_MKLDNN define

### CODE REVIEW

**Comment by larroy** [comment]
LGTM

File: CMakeLists.txt [hunk 1]
```diff
@@ -170,7 +172,6 @@ if(USE_MKL_IF_AVAILABLE)
     list(APPEND mxnet_LINKER_LIBS iomp5)
   else()
     message(STATUS " MKL not found")
-  endif()
```
Comment (marcoabreu) [inline CMakeLists.txt]: is this removal intentional? There's no matching if block that has been removed

**Comment by marcoabreu** [comment]
@zheng-da could you verify?

**Comment by pengzhao-intel** [comment]
LGTM :+1:
